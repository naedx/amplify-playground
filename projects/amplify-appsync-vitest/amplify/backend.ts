import { defineBackend } from '@aws-amplify/backend';
import { Stack } from 'aws-cdk-lib';
import { defineCustomData } from './data/custom-resource';
import { data } from './data/resource';

/**
 * @see https://docs.amplify.aws/react/build-a-backend/ to add storage, functions, and more
 */
const backend = defineBackend({
  // auth,
  data,
});

export type DefinedBackendResourcesType = typeof backend;

// This function manually creates the DataSources > Resolvers
defineCustomData<DefinedBackendResourcesType>(Stack.of(backend.data), backend);
