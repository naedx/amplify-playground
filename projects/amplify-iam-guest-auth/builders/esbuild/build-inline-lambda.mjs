/* eslint-disable */
// This file configures esbuild for the AppSync typescript resolvers
// Task: watch:ts-resolvers

import { build, context } from 'esbuild';
import eslint from 'esbuild-plugin-eslint';

const catchWarningsAndErrorsPlugin = {
  name: 'catch-warnings-and-errors',
  setup(build) {
    let count = 0;
    build.onEnd(result => {
      count++;

      if (result?.warnings?.length > 0) {
        console.log(`Warnings [${count}]:`, result?.warnings);
      }

      if (result?.errors?.length > 0) {
        console.log(`Errors [${count}]:`, result?.errors);
      }

    });
  },
};

const config = {
  sourcemap: 'inline',
  sourcesContent: false,

  format: 'cjs',
  target: 'esnext',
  platform: 'node',
  external: [
    'aws-lambda',
    '@aws-sdk/*',
    '@aws-sdk/client-s3',
    '@aws-sdk/client-appsync',
    '@aws-sdk/s3-request-presigner',
    '@aws-sdk/client-rekognition'
  ],
  outExtension: {
    '.js': '.cjs'
  },
  outbase: 'amplify/data',
  outdir: 'amplify/data',
  entryPoints: [
    'amplify/data/features/**/*.lambda.ts',
    // 'amplify/data/features/**/*.lambda/*.ts'
  ],
  bundle: true,
  plugins: [
    eslint({
      useEslintrc: false,
      overrideConfigFile: 'amplify/data/.eslintrc.lambda.json'
    })
  ],
};

let mode = 'build';

if (process.argv.length > 2 && process.argv[2] === '--watch') {
  mode = 'watch';
}

if (mode === 'build') {
  await build(config);
}
else {
  async function watch() {
    config.plugins = [...config.plugins, catchWarningsAndErrorsPlugin];

    let ctx = await context(config);

    await ctx.watch();

    // await ctx.dispose();

    console.log(`Watching...`);
  }

  await watch();

}
