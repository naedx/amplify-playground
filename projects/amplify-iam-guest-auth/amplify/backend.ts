import { defineBackend } from '@aws-amplify/backend';
import { Stack } from 'aws-cdk-lib';
import { auth } from './auth/resource';
import { defineCustomData } from './data/custom-resource';
import { data } from './data/resource';

/**
 * @see https://docs.amplify.aws/react/build-a-backend/ to add storage, functions, and more
 */
const backend = defineBackend({
  auth,
  data,
});

export type DefinedBackendResourcesType = typeof backend;

defineCustomData(Stack.of(backend.data), backend);
