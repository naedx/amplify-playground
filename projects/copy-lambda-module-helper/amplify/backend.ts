import { defineBackend } from '@aws-amplify/backend';
import {
  Function as LambdaFunction,
  Runtime as LambdaRuntime,
} from 'aws-cdk-lib/aws-lambda';
import path from 'path';
import { fromAssetHelper, TranspilerOptions } from './backend.utils';

const backend = defineBackend({
  // auth,
  // data,
  // storage
});

const stack = backend.createStack('MyCustomStack');
// const stack = Stack.of(backend.storage.resources.bucket); // or something like this

const myLambda = new LambdaFunction(stack, 'MyCustomFunction', {
  handler: 'index.handler',
  code: fromAssetHelper(
    path.resolve('amplify/functions/my-custom-function.lambda.cjs')
  ),
  runtime: LambdaRuntime.NODEJS_20_X,
});
