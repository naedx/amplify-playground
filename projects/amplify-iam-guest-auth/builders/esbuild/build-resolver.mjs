/* eslint-disable */
// This file configures esbuild for the AppSync typescript resolvers
// Task: watch:ts-resolvers

import { build, context } from 'esbuild';
import eslint from 'esbuild-plugin-eslint';

const files = 'amplify/data/resolvers-ts/**/*.ts';

const catchWarningsAndErrorsPlugin = {
  name: 'catch-warnings-and-errors',
  setup(build) {
    let count = 0;
    build.onEnd(result => {
      count++;

      if (result?.warnings?.length > 0) {
        console.log(`Warnings [${count}]:`, result?.warnings);
      }

      if (result?.errors?.length > 0) {
        console.log(`Errors [${count}]:`, result?.errors);
      }

    });
  },
};

const config = {
  sourcemap: 'inline',
  sourcesContent: false,

  format: 'esm',
  target: 'esnext',
  platform: 'node',
  external: [
    '@aws-appsync/utils',
    '@aws-sdk/client-s3',
    '@aws-sdk/s3-request-presigner'
  ],
  outbase: 'amplify/data',
  outdir: 'amplify/data',
  entryPoints: [
    'amplify/data/common/**/*.ts',
    'amplify/data/features/**/*.resolver.ts'
  ],

  bundle: true,
  plugins: [
    eslint({
      useEslintrc: false,
      overrideConfigFile: 'amplify/data/.eslintrc.resolver.json'
    })
  ],
};

let mode = 'build';

if (process.argv.length > 2 && process.argv[2] === '--watch') {
  mode = 'watch';
}

if (mode === 'build') {
  await build(config);
}
else {
  async function watch() {
    config.plugins = [...config.plugins, catchWarningsAndErrorsPlugin];

    let ctx = await context(config);

    await ctx.watch();

    // await ctx.dispose();

    console.log(`Watching... ${files.toString()}`);
  }

  await watch();

}
