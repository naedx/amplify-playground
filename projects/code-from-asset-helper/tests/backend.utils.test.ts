

import path from 'path';
import { describe, expect, test } from 'vitest';
import { esbuildBuilding, esLinting, tsTranspiling } from '../lib/backend.utils';

import { BuildOptions as ESBuildOptions } from 'esbuild';

import fs from 'fs';

describe('appsync builder', () => {

  test('resolver should fail linting', async () => {

    const sourceFilePath = path.resolve('./tests/create-report.fail.resolver.ts');

    expect(async () => {
      const lintResult = await esLinting(sourceFilePath, path.resolve('./tests/tsconfig.json'));

    }).rejects.toThrow('2 linting error(s)');

  });

  test('esbuild should create output', async () => {

    const sourceFilePath = path.resolve('./tests/create-report.resolver.ts');

    const outputDir = path.resolve(`./tests/built-assets/asset.${getRandomInt(10000, 99999)}`);

    esLinting(sourceFilePath, path.resolve('./tests/tsconfig.json'));

    //AppSync function build
    const esBuildOptions: ESBuildOptions = {
      sourcemap: 'inline',
      sourcesContent: false,
      format: 'esm',
      target: 'esnext',
      platform: 'node',
      external: [
        '@aws-appsync/utils',
        '@aws-sdk/client-s3',
        '@aws-sdk/s3-request-presigner',
      ],
      bundle: true,
    };

    const outFile = await esbuildBuilding({
      sourceFilePath,
      outputFilePath: path.join(outputDir, path.basename(sourceFilePath).replace('.ts', '.js')),
      buildOptions: esBuildOptions
    });

    expect(outFile).not.toBeNull();

    const assetCreated = fs.existsSync(outFile);

    expect(assetCreated).toBe(true);

  });

  test('tsTranspiling should create output', async () => {

    const sourceFilePath = path.resolve('./tests/create-report.resolver.ts');

    const outputDir = path.resolve(`./tests/built-assets/asset.${getRandomInt(10000, 99999)}`);

    if (!fs.existsSync(outputDir)) fs.mkdirSync(outputDir);

    esLinting(sourceFilePath, path.resolve('./tests/tsconfig.json'));

    const outFile = tsTranspiling(sourceFilePath, path.join(outputDir, path.basename(sourceFilePath).replace('.ts', '.js')))

    expect(outFile).not.toBeNull();

    const assetCreated = fs.existsSync(outFile);

    expect(assetCreated).toBe(true);

  });

});

function getRandomInt(min: number, max: number) {
  const minCeiled = Math.ceil(min);
  const maxFloored = Math.floor(max);
  return Math.floor(Math.random() * (maxFloored - minCeiled) + minCeiled); // The maximum is exclusive and the minimum is inclusive
}

