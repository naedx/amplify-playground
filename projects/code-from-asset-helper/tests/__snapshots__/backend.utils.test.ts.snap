// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`appsync builder > esbuild should detect errors in nested files 1`] = `
[Error: 2 nested file(s) have errors. 

 
The AppSync source file was not built successfully: tests/helpers.ts

3 linting error(s) found:: 

:: Lint Rule Error ::
Rule: @typescript-eslint/no-explicit-any
Severity: 2
Message: Unexpected any. Specify a different type.
Location: tests/helpers.ts:6:13 to 6:16
Suggestion(s): 
  Use \`unknown\` instead, this will force you to explicitly, and safely assert the type is correct. Col 256-259 'unknown'
  Use \`never\` instead, this is useful when instantiating generic type parameters that you don't need to know the type of. Col 256-259 'never'.

:: Lint Rule Error ::
Rule: @typescript-eslint/no-unsafe-member-access
Severity: 2
Message: Unsafe member access .sub on an \`any\` value.
Location: tests/helpers.ts:8:28 to 8:31


:: Lint Rule Error ::
Rule: @typescript-eslint/no-unsafe-member-access
Severity: 2
Message: Unsafe member access .username on an \`any\` value.
Location: tests/helpers.ts:8:44 to 8:52








The AppSync source file was not built successfully: tests/utils.fail.resolver.ts

3 linting error(s) found:: 

:: Lint Rule Error ::
Rule: @aws-appsync/no-async
Severity: 2
Message: Async functions are not supported.
Location: tests/utils.fail.resolver.ts:1:8 to 4:2


:: Lint Rule Error ::
Rule: @typescript-eslint/require-await
Severity: 2
Message: Async function 'addition' has no 'await' expression.
Location: tests/utils.fail.resolver.ts:1:8 to 1:31
Suggestion(s): 
  Remove 'async'. Col 7-69 'function addition(a: number, b: number): number'.

:: Lint Rule Error ::
Rule: @typescript-eslint/no-unused-vars
Severity: 2
Message: 'c' is assigned a value but never used.
Location: tests/utils.fail.resolver.ts:2:9 to 2:10





]
`;

exports[`appsync builder > full asset helper should detect lint errors 1`] = `
[Error: 2 nested file(s) have errors. 

 
The AppSync source file was not built successfully: tests/helpers.ts

3 linting error(s) found:: 

:: Lint Rule Error ::
Rule: @typescript-eslint/no-explicit-any
Severity: 2
Message: Unexpected any. Specify a different type.
Location: tests/helpers.ts:6:13 to 6:16
Suggestion(s): 
  Use \`unknown\` instead, this will force you to explicitly, and safely assert the type is correct. Col 256-259 'unknown'
  Use \`never\` instead, this is useful when instantiating generic type parameters that you don't need to know the type of. Col 256-259 'never'.

:: Lint Rule Error ::
Rule: @typescript-eslint/no-unsafe-member-access
Severity: 2
Message: Unsafe member access .sub on an \`any\` value.
Location: tests/helpers.ts:8:28 to 8:31


:: Lint Rule Error ::
Rule: @typescript-eslint/no-unsafe-member-access
Severity: 2
Message: Unsafe member access .username on an \`any\` value.
Location: tests/helpers.ts:8:44 to 8:52








The AppSync source file was not built successfully: tests/utils.fail.resolver.ts

3 linting error(s) found:: 

:: Lint Rule Error ::
Rule: @aws-appsync/no-async
Severity: 2
Message: Async functions are not supported.
Location: tests/utils.fail.resolver.ts:1:8 to 4:2


:: Lint Rule Error ::
Rule: @typescript-eslint/require-await
Severity: 2
Message: Async function 'addition' has no 'await' expression.
Location: tests/utils.fail.resolver.ts:1:8 to 1:31
Suggestion(s): 
  Remove 'async'. Col 7-69 'function addition(a: number, b: number): number'.

:: Lint Rule Error ::
Rule: @typescript-eslint/no-unused-vars
Severity: 2
Message: 'c' is assigned a value but never used.
Location: tests/utils.fail.resolver.ts:2:9 to 2:10





]
`;
